src/main/scala/com/atomist/rug/function/github/GitHubFunction.scala

```diff
@@ -1,70 +1,56 @@
 package com.atomist.rug.function.github
 
-import java.io.InputStream
-import java.text.SimpleDateFormat
 import java.time.{OffsetDateTime, ZoneId}
 import java.util.Date
 
 import com.atomist.rug.runtime.Rug
-import com.atomist.source.git.GitHubServices
-import com.fasterxml.jackson.annotation.JsonInclude.Include
-import com.fasterxml.jackson.databind.{DeserializationFeature, ObjectMapper, SerializationFeature}
-import com.fasterxml.jackson.datatype.jdk8.Jdk8Module
-import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
-import com.fasterxml.jackson.module.scala.DefaultScalaModule
-import com.fasterxml.jackson.module.scala.experimental.ScalaObjectMapper
+import com.atomist.source.git.github.GitHubServices
 
-trait GitHubFunction
-  extends Rug {
+trait GitHubFunction extends Rug {
 
   /**
     * Sanitize a token.
     */
   def safeToken(token: String): String =
-    if (token != null) {
+    if (token != null)
       token.charAt(0) + ("*" * (token.length() - 2)) + token.last
-    } else {
+    else
       null
-    }
 
   def gitHubServices(token: String, apiUrl: String): GitHubServices =
-    apiUrl match {
-      case url: String => GitHubServices(token, url)
-      case _ => GitHubServices(token)
-    }
+    GitHubServices(token, Option(apiUrl))
```

Comment:
I see now, you've created what is in effect an alternative constructor. Why not just use a case class and define multiple `apply` methods in the companion object?